### PR TITLE
refactor(types): add restricted types as requires for composites

### PR DESCRIPTION
**Pull request by mbroadst**

### CODE REVIEW

File: lib/frames.js [hunk 1]
```diff
@@ -207,7 +237,7 @@ frames.TransferFrame = defineFrame(FrameType.AMQP, {
     { name: 'messageFormat', type: 'message-format', default: 0 },
     { name: 'settled', type: 'boolean' },
     { name: 'more', type: 'boolean', default: false },
-    { name: 'rcvSettleMode', type: 'ubyte', default: 0 }, // @spec type: 'receiver-settle-mode` => `boolean` choice
+    { name: 'rcvSettleMode', type: 'ubyte', requires: receiverSettleMode, default: 0 },
```
Comment (noodlefrenzy) [inline lib/frames.js]: `default: 'auto'`

File: lib/frames.js [hunk 2]
```diff
@@ -126,6 +126,38 @@ function defineFrame(type, definition) {
   return Frame;
 }
 
+// restricted type helpers
+function role(value) {
```
Comment (noodlefrenzy) [inline lib/frames.js]: Man, it really _feels_ like there should be a way to easily factor this into common code. I just can't see what it is (I keep seeing how I'd write it in C#).

Comment (mbroadst) [inline lib/frames.js]: yeah I mean, most of these are just "choices", so we could instead set `requires: { mapping: 'of', objects: 'here' }`, but I figured doing it with a function was very easy for now - and doesn't cost too much

File: lib/types/terminus.js [hunk 3]
```diff
@@ -1,17 +1,40 @@
 'use strict';
 var types = require('../types'),
+    errors = require('../errors'),
     Terminus = module.exports = {};
 
+function terminusDurability(value) {
```
Comment (noodlefrenzy) [inline lib/types/terminus.js]: If you want to be consistent, we should only accept ['none', 'configuration', 'unsettled-state'] and default to 'none' below.

Comment (mbroadst) [inline lib/types/terminus.js]: well I've been trying to be consistent in terms of supporting the path where you might be using our constants still (like, the whole codebase), while also supporting string versions. But yeah I'll default it to `none` in the field definition

**Pull request closed without merging**